utils/writeable/src/utf.rs

```diff
@@ -0,0 +1,89 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::{impl_display_with_writeable, LengthHint, Writeable};
+
+use core::fmt;
+
+/// Implements [`Writeable`] for [`&[u8]`] according to the [WHATWG Encoding Standard](
+/// https://encoding.spec.whatwg.org/#utf-8-decoder).
+#[derive(Debug)]
+#[allow(clippy::exhaustive_structs)] // newtype
+pub struct PotentiallyInvalidUtf8<'a>(pub &'a [u8]);
```

Comment:
Nit: I would rather have this type in its own crate, with an optional dependency in one or the other direction that includes this impl. See #3546

Reply:
I don't, this feels like a core writeable concern, and I don't like a million tiny crates.

Reply:
I'm okay making `Writeable` the crate where we put the types discussed in #3546. Are you okay making this struct the replacement for `UnvalidatedStr`? If so, I'd prefer it to be an unsized type.

Reply:
There's no conclusion in https://github.com/unicode-org/icu4x/issues/3546 that would let me move ahead here.

Reply:
I don't think `UnvalidatedStr` should live in `Writeable`, as then `Writeable` would gain an optional `serde` dependency. I think for now we should have two types, and unify them once we have a conclusion on #3546. I can make these `#[doc(hidden)]`.

Reply:
Re unsized, I'd rather not as they'd need a constructor with a transmute.

Reply:
I prefer marking the types as experimental in the docs over `#[doc(hidden)]`. Maybe add an experimental feature.

Reply:
We don't use experimental features on utils.

Reply:
They could also live privately in `icu_capi` for now.

Reply:
I don't see a scenario where these types end up being in this crate given the `serde` dependency as you noted, so it seems prudent to keep them internal in `icu_capi` and then we can merge them into the new crate when we agree on its name.